Emby.Server.Implementations/SystemManager.cs

```diff
@@ -72,16 +74,30 @@ public SystemInfo GetSystemInfo(HttpRequest request)
         };
     }
 
-    /// <inheritdoc />
+    /// <summary>
+    /// Gets public system info of the server.
+    /// LocalAddress is the server URL that can also be used by applications not on the server, for instance
+    /// a cast receiver device. In case the request has a localhost url, LocalAddress will be set to a
+    /// server Private network address (IPv4) or Unique Local Address (ULA, IPv6) which is valid for all
+    /// devices on the LAN.
+    /// </summary>
+    /// <param name="request">Where this request originated.</param>
+    /// <returns>PublicSystemInfo.</returns>
     public PublicSystemInfo GetPublicSystemInfo(HttpRequest request)
     {
+        IPAddress host = request.HttpContext.Connection.RemoteIpAddress ?? IPAddress.None;
+        string localAddress =
+            Network.IPv4RFC5735Loopback.Contains(host) || Network.IPv6RFC4291Loopback.Contains(host)
+            ? _applicationHost.GetSmartApiUrl(NetworkManager.ServerLanAddress)
+            : _applicationHost.GetSmartApiUrl(host);
```

Comment:
This actually breaks `EnablePublishedServerUriByRequest` and then there's also a discrepancy between GetSystemInfo and GetPublicSystemInfo.

The check should probably be put into `public string GetSmartApiUrl(HttpRequest request)`

Reply:
Done!